pkg/client/unversioned/clientcmd/config.go

```diff
@@ -75,8 +75,9 @@ func (o *PathOptions) GetLoadingPrecedence() []string {
 	if envVarFiles := o.GetEnvVarFiles(); len(envVarFiles) > 0 {
 		return envVarFiles
 	}
-
-	return []string{o.GlobalFile}
+	result := []string{o.GlobalFile}
+	result = append(result, loadConfigDir(o.GlobalFile+".d")...)
```

Comment:
This sets off a spidey sense for me... Should we have a separate field `GlobalFileDir` or similar? And then we would set it to `RecommendedHomeFileDir` in NewDefaultPathOptions?